bundle/edu.gemini.spdb.reports.collection/src/main/scala/edu/gemini/dbTools/ephemeris/NonSiderealObservationFunctor.scala

```diff
@@ -0,0 +1,30 @@
+package edu.gemini.dbTools.ephemeris
+
+import edu.gemini.pot.sp.{ISPProgram, ISPNode}
+import edu.gemini.pot.spdb.{IDBDatabaseService, DBAbstractQueryFunctor}
+
+import java.security.Principal
+import java.util.{Set => JSet}
+
+
+import scalaz._, Scalaz._
+
+/** An ODB query functor that finds all scheduleable non-sidereal
+  * observations.
+  */
+object NonSiderealObservationFunctor {
+  def query(db: IDBDatabaseService, users: JSet[Principal]): List[NonSiderealObservation] =
+    new NonSiderealObservationFunctor |> (f => db.getQueryRunner(users).queryPrograms(f).results)
+}
+
+private class NonSiderealObservationFunctor extends DBAbstractQueryFunctor{
+  import NonSiderealObservation.findScheduleableIn
+
+  var results: List[NonSiderealObservation] = Nil
```

Comment:
Right, the ODB query runner calls the `execute` method repeatedly, once for each program in the database.  The so-called functor has to mutate to build up its result.  This could totally be reworked to avoid mutation at the expense of rewriting all the existing functors.